.github/workflows/test-default.yaml

```diff
@@ -7,7 +7,22 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
+      - name: Cloning repository
+        uses: actions/checkout@v4
 
-      - name: 'Default'
+      - name: Setting up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Installing Node.js packages
+        run: npm ci
+
+      - name: Running tests
         run: |
-          echo 'we require a job named "test", but not all pull requests trigger an action. So we created this small silly job named "test" that will always pass, just to make sure nothing hangs as a result of not triggering an actual job.'
+          npx gulp updateTestResources
+          npx gulp lintNode
+          npx gulp lintYaml
+          npm run test:platform
+          npm run test:playground
```

Comment:
These two lines are also in `test-platform.yaml` and `test-playground.yaml` - let's avoid duplication here. We can probably just remove the conditionals on these other files, since these tests are relatively fast and it's better to be safe than sorry here

(basically, replace the `on:` block in these three files with just `on: [dispatch_workflow, push, pull_request]`